### ISSUE TITLE
App crashes when "list" is changed in VirtuosoGrid

### ISSUE THREAD

**Issue by mcabs3** [petyosi/react-virtuoso#24]
first off, this library is great and exactly what I need!

I am using the `VirtuosoGrid` component, I supply it an array of small objects (from redux). I am trying to allow a user to use a text input to search and filter the boxes in the grid, so I have the text input text filter the array that is passed to this grid, when the list is "filtered" my react app crashes because a component that did have one of the "items" is now getting undefined.

I'm probably missing something simple. But it seems like this should just _work_?

```jsx
// user typed in "mi"
const searchText = 'z';
const list = [{a: 'Mike'}, {a: 'Sam'}....... {a: 'Michael'}];

// some logic to filter the list based on text input
const filteredList = list.filter(item => item.a === searchText);

// down way below in renderland
<VirtuosoGrid
  totalCount={filteredList.length}
  item={idx => <MyComponent item={filteredList[idx]} />}
/>

```

**Comment by petyosi** [comment]
Hey, @mcabs3 that should work from what I see, but you may have found a bug in the rendering mechanism. Can you reproduce it in stackblitz or codesandbox? I will take a look. 

Cheers,

**Comment by mcabs3** [comment]
@petyosi Can do: https://codesandbox.io/embed/silly-mendeleev-8j51k

Try searching successfully, and then try a random set of characters to where nothing would return from the filter.

**Comment by petyosi** [comment]
Thank you! Reproduced the problem on my side, confirming it as a bug. Will figure out a fix and publish a new version this week.

**Comment by mcabs3** [comment]
Thank you, I appreciate the quick attention!

**Comment by petyosi** [comment]
Hey, just published v0.10 which includes a fix for the above. Forked your sandbox and updated the version - it seems to work fine. 

https://codesandbox.io/s/serene-fog-wyx8v 

Cheers,

**Comment by mcabs3** [comment]
Sounds good, I will give it a try on my actual app as well. Thanks for squashing this so quick!
